libusb/os/sunos_usb.c

```diff
@@ -1361,6 +1361,7 @@ solaris_submit_ctrl_on_default(struct libusb_transfer *transfer)
 	usbi_dbg("Done: ctrl data bytes %d", ret);
 
 	/* sync transfer handling */
+	usbi_mutex_unlock(&LIBUSB_TRANSFER_TO_USBI_TRANSFER(transfer)->lock);
 	ret = usbi_handle_transfer_completion(LIBUSB_TRANSFER_TO_USBI_TRANSFER(transfer),
```

Comment:
Where is the corresponding lock? Why is unlock not called on error conditions. Does the lock need to be re-obtained after calling ```usbi_handle_transfer_completion()```?

Just want to make sure I understand what is wrong.

Reply:
This lock is taken in libusb_submit_transfer(), which calls usbi_backend.submit_transfer() , which in its turn calls solaris_submit_ctrl_on_default(). 

usbi_handle_transfer_completion() takes its own lock on itransfer->lock, so without this we hang.

Reply:
We return immediately after usbi_handle_transfer_completion. Perhaps, we should re-obtain lock after calling usbi_handle_transfer_completion(), have to check.